recipe/meta.yaml

```diff
@@ -1,82 +1,75 @@
 {% set name = "anyio" %}
-{% set version = "3.5.0" %}
+{% set version = "4.2.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  - folder: dist
-    url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-    sha256: a0aeffe2fb1fdf374a8e4b471444f0f3ac4fb9f5a5b542b48824475e0042a5a6
-  - folder: src
-    url: https://github.com/agronholm/{{ name }}/archive/{{ version }}.tar.gz
-    sha256: 55a37ea9eb2092d435af148b42a10005ee62ec7eb7d7a12d73c88e2b8aca5e38
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: e1875bb4b4e2de1669f4bc7869b6d3f54231cdced71605e6e64c9be77e3be50f
+  patches: # [unix]
+    - patches/0001-Modify-unix-socket-tests-to-use-stdlib-tempdirs.patch # [unix]
 
 build:
   number: 0
-  # trigger: 0
-  script: cd dist && {{ PYTHON }} -m pip install . -vv --no-deps
-  skip: true  # [py<36]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: true  # [py<38]
 
 requirements:
   build:
-    - m2-patch                               # [win]
-    - patch                                  # [not win]
+    - patch # [unix]
   host:
     - pip
     - python
-    - setuptools-scm >=3.4
-    - setuptools >=42
+    - setuptools-scm >=6.4
+    - setuptools >=64
     - toml
-    - wheel >=0.29.0
+    - wheel
   run:
-    - contextvars  # [py<37]
-    - dataclasses  # [py<37]
+    - python
+    - exceptiongroup >=1.0.2  # [py<311]
     - idna >=2.8
-    - python >=3.6.2
     - sniffio >=1.1
-    - typing_extensions  # [py<38]
+    - typing_extensions >=4.1  # [py<311]
   run_constrained:
-    - trio >=0.16
+    - trio >=0.23
+    - uvloop >=0.17
 
 test:
   source_files:
-    - src/tests
+    - tests
+    - pyproject.toml
   requires:
-    - contextlib2  # [py<37]
-    - coverage >=4.5
-    - trio
+    - exceptiongroup >=1.2.0
     - hypothesis >=4.0
-    - mock >=4
     - pip
-    - pytest >=6.0
-    - pytest-cov
+    - psutil >=5.9
+    - pytest >=7.0
     - pytest-mock >=3.6.1
-    - toml
+    - tomli
+    - trio
     - trustme
-    # won't be available on pypy, or windows, any time soon
-    - uvloop  # [unix and python_impl == 'cpython']
-    - ruamel_yaml  # [win]
+    - uvloop >=0.17  # [unix]
   imports:
     - anyio
   commands:
-    - pip check || 1     # [win]
-    - pip check || true  # [unix]
-  # To check if downstream packages run without errors
-  downstreams:
-    # Testing jupyter_server sometimes fails in win platforms
-    - jupyter_server  # [unix]
```

Comment:
I would say "yes" and recommend testing it every time we update `anyio`. But then comment it out before merging.
jupyter_server often fails during tests because of the async code.

Reply:
I added back the downstream part (now commented to avoid circular dependencies).
Tests of jupyter_server passed. 
Note: most packages depending on anyio in main have an upper bound set.